inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_opset1_to_legacy.cpp

```diff
@@ -46,26 +45,103 @@
-    ngraph::pass::Manager OpSet1ToLegacy;
+    ngraph::pass::Manager manager;
     std::vector<std::shared_ptr<ngraph::pass::PassBase> > transforms;
 
-#define NGRAPH_PASS(NAME, NAMESPACE) transforms.push_back(OpSet1ToLegacy.register_pass<NAMESPACE::NAME>());
-#include <transformations/convert_opset1_to_legacy/convert_opset1_to_legacy_tbl.hpp>
-#undef NGRAPH_PASS
+    manager.register_pass<ngraph::pass::ConstantFolding>();
+
+    // List if Decomposition and Conversion transformations that can be
+    // applied simultaneously in a single graph traversal
+    auto decomp = manager.register_pass<ngraph::pass::GraphRewrite>();
+    decomp->add_matcher<ngraph::pass::ConvertBroadcastToTiles>();
+    decomp->add_matcher<ngraph::pass::ConvertReduceMeanToPooling>();
+    decomp->add_matcher<ngraph::pass::ConvertReduceMaxToPooling>();
+    decomp->add_matcher<ngraph::pass::ConvertReduceSumToPooling>();
+    decomp->add_matcher<ngraph::pass::ConvertMod>();
+    decomp->add_matcher<ngraph::pass::ConvertMinimum>();
+    decomp->add_matcher<ngraph::pass::ConvertSubtract>();
+    decomp->add_matcher<ngraph::pass::ConvertDivide>();
+    decomp->add_matcher<ngraph::pass::ConvertNegative>();
+    decomp->add_matcher<ngraph::pass::ConvertDepthToSpace>();
+    decomp->add_matcher<ngraph::pass::ConvertSpaceToDepth>();
+    decomp->add_matcher<ngraph::pass::ConvertConvolution>();
+    decomp->add_matcher<ngraph::pass::ConvertGroupConvolution>();
+    decomp->add_matcher<ngraph::pass::ConvertDeconvolution>();
+    decomp->add_matcher<ngraph::pass::ConvertGroupDeconvolution>();
+    decomp->add_matcher<ngraph::pass::BatchNormDecomposition>();
+    decomp->add_matcher<ngraph::pass::ConvertMatMulToFCorGemm>();
+    decomp->add_matcher<ngraph::pass::PullTransposeThroughFQUp>();
+    decomp->set_name("ngraph::pass::Decompositions");
+
+    // CF is required after all decompositions
+    manager.register_pass<ngraph::pass::ConstantFolding>();
+
+    // LinOpSequenceFusion must be executed after all decompositions
+    manager.register_pass<ngraph::pass::LinOpSequenceFusion>();
+
+    // Convolution/Deconvolution/FullyConnected fusions
+    auto fusion = manager.register_pass<ngraph::pass::GraphRewrite>();
+    fusion->add_matcher<ngraph::pass::ConvAddFusion>();
+    fusion->add_matcher<ngraph::pass::ConvMultiplyFusion>();
+    fusion->add_matcher<ngraph::pass::DeconvAddFusion>();
+    fusion->add_matcher<ngraph::pass::FullyConnectedBiasFusion>();
+    fusion->set_name("ngraph::pass::Fusions");
+
+    // CF is required after fusions
+    manager.register_pass<ngraph::pass::ConstantFolding>();
+
+    // List of passes that convert opset1 operations to legacy
+    // plus transformations that are required by InferenceEngine
+    // All this transformations can be executed simultaneously
+    auto anchor = manager.register_pass<ngraph::pass::GraphRewrite>();
+    anchor->add_matcher<ngraph::pass::ReshapeFullyConnected>();
```

Comment:
can `manager` merge 2 or more `GraphRewrite` passes? and use single pass through the graph instead of 2

Reply:
If you want to to merge two GraphRewrite passes means that you can take MatcherPasses from them and combine manually.
pass::Manager responsible only for execution not combining. I think that in this case GraphRewrite `add_matcher` should support another GrpahRewrite instance as input and take matcher passes from it. But I still think that it's no good idea to merge two GraphRewrite passes